### PR TITLE
Metadata updates

### LINKED ISSUE #63: Take up changes from review
Here's a diff with some of them:

```diff
diff --git a/collections/modis/modis-17A2HGF-061/template.json b/collections/modis/modis-17A2HGF-061/template.json
index 8ddfcb6..76a142f 100644
--- a/collections/modis/modis-17A2HGF-061/template.json
+++ b/collections/modis/modis-17A2HGF-061/template.json
@@ -37,7 +37,7 @@
             "roles": [
                 "data"
             ],
-            "title": "hdf data",
+            "title": "Source data containing all bands",
             "eo:bands": [
                 {
                     "name": "Gpp_500m",
@@ -58,12 +58,12 @@
             "roles": [
                 "metadata"
             ],
-            "title": "FGDC Metdata"
+            "title": "Federal Geographic Data Committee (FGDC) Metadata"
         },
         "Gpp_500m": {
             "type": "image/tiff; application=geotiff; profile=cloud-optimized",
-            "title": "Gpp_500m",
-            "description": "Gross Primary Productivity",
+            "title": "Gross Primary Productivity",
+            "description": "better description",
             "eo:bands": [
                 {
                     "name": "Gpp_500m",
@@ -117,8 +117,6 @@
             "raster:bands": [
                 {
                     "data_type": "uint8",
-                    "scale": null,
-                    "unit": null,
                     "spatial_resolution": 500
                 }
             ],
@@ -127,11 +125,11 @@
             ]
         }
     },
-    "msft:short_description": "MODIS Gross Primary Productivity Gap-Filled 8-Day L4 Global 500 m SIN Grid",
+    "msft:short_description": "Make this more meaningful and descriptive",
     "msft:storage_account": "modiseuwest",
     "msft:container": "modis-061",
     "msft:group_id": "modis",
-    "title": "MODIS Gross Primary Productivity Gap-Filled 8-Day L4 Global 500 m SIN Grid",
+    "title": "MODIS Gross Primary Productivity Gap-Filled 8-Day",
     "extent": {
         "spatial": {
             "bbox": [
@@ -169,17 +167,22 @@
             "roles": [
                 "producer",
                 "licensor",
-                "processor",
-                "host"
+                "processor"
             ],
             "url": "https://lpdaac.usgs.gov/"
         },
         {
-            "name": "Microsoft",
+            "name": "Element 84",
             "roles": [
-                "host",
                 "processor"
             ],
+            "url": ""
+        },
+        {
+            "name": "Microsoft",
+            "roles": [
+                "host"
+            ],
             "url": "https://planetarycomputer.microsoft.com"
         }
     ],
@@ -194,7 +197,7 @@
     },
     "assets": {
         "thumbnail": {
-            "href": "https://ai4edatasetpublitcassets.blob.core.windows.net/assets/pc_thumbnails/modis-17A2HGF-061.png",
+            "href": "https://ai4edatasetpublicassets.blob.core.windows.net/assets/pc_thumbnails/modis-17A2HGF-061.png",
             "type": "image/png",
             "title": "MODIS Gross Primary Productivity Gap-Filled 8-Day L4 Global 500 m SIN Grid thumbnail",
             "roles": [
```

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):**
- Closes #63 

**Description:** Update many of the link titles and other metadata.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Example STAC Catalog has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: examples/modis-006/modis-10A1-006/collection.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     {
       "rel": "describedby",
       "href": "https://doi.org/10.5067/MODIS/MOD10A1.006",
-      "title": "Hall, D. K., V. V. Salomonson, and G. A. Riggs. 2016. MODIS/Terra Snow Cover Daily L3 Global 500m Grid. Version 6. Boulder, Colorado USA: NASA National Snow and Ice Data Center Distributed Active Archive Center."
+      "title": "MODIS/Terra Snow Cover Daily L3 Global 500m SIN Grid, Version 61"
     },
     {
       "rel": "cite-as",
```
Comment (pjhartzell) [inline examples/modis-006/modis-10A1-006/collection.json:19]: missing the title in doi cite-as links for all modis-006 collections.

File: examples/modis-006/modis-10A1-006/MOD10A1.A2022029.h10v05.006.2022031045818/MOD10A1.A2022029.h10v05.006.2022031045818.json [hunk 2]
```diff
@@ -67,45 +67,15 @@
...
-        {
-          "name": "orbit_pnt",
-          "description": "Pointer to the orbit of the swath mapped into each grid cell. For details, see Section 1.1.2 below."
-        },
-        {
-          "name": "granule_pnt",
-          "description": "Pointer for identifying the swath mapped into each grid cell. For details, see the section 'Using the granule_pnt' below."
-        }
-      ],
+      "title": "Source data containing all bands",
```
Comment (pjhartzell) [inline examples/modis-006/modis-10A1-006/MOD10A1.A2022029.h10v05.006.2022031045818/MOD10A1.A2022029.h10v05.006.2022031045818.json:70]: Can't figure out how to comment on the file generally. It looks like you are placing bitfield "flags" in the properties. Interesting.

**Review by pjhartzell — COMMENTED** [review]
Do most of the items not require projection info? I only see it in two of the many example item JSON files.

File: examples/modis-006/modis-10A1-006/MOD10A1.A2022029.h10v05.006.2022031045818/MOD10A1.A2022029.h10v05.006.2022031045818.json (hunk 2 above)
Comment (gadomski) [reply examples/modis-006/modis-10A1-006/MOD10A1.A2022029.h10v05.006.2022031045818/MOD10A1.A2022029.h10v05.006.2022031045818.json:70]: Do you mean `file:values` on the asset, e.g. [here](https://github.com/stactools-packages/modis/blob/33d2f6868720aa11b5cda44fcbd719b2c4065895/examples/modis-061/modis-10A2-061/MOD10A2.A2022033.h09v05.061.2022042050729/MOD10A2.A2022033.h09v05.061.2022042050729.json#L158-L219)?

Comment (pjhartzell) [reply examples/modis-006/modis-10A1-006/MOD10A1.A2022029.h10v05.006.2022031045818/MOD10A1.A2022029.h10v05.006.2022031045818.json:70]: No. It's an 006 thing, so no matter. I commented on the wrong file, I guess. See here: https://github.com/stactools-packages/modis/blob/issues/63-metadata-updates/src/stactools/modis/fragments/21A2/006/item-properties.json, which gets pushed into some example jsons: https://github.com/stactools-packages/modis/blob/issues/63-metadata-updates/examples/modis-006/modis-21A2-006/MOD21A2.A2005361.h10v04.006.2018286181506/MOD21A2.A2005361.h10v04.006.2018286181506.json

Comment (gadomski) [reply examples/modis-006/modis-10A1-006/MOD10A1.A2022029.h10v05.006.2022031045818/MOD10A1.A2022029.h10v05.006.2022031045818.json:70]: Yeah those were leftover from the first pass at this package, I would say they're _not_ the recommended way of doing things.

**Review by pjhartzell — APPROVED** [review]
LGTM.

**Pull request merged**